### DISCUSSION TITLE
Bookmark and Share Posts

### DISCUSSION THREAD

**Discussion by Shubs21** [deso-protocol/dips#38] [Medium CIPs]
Implementing the 'Bookmark Post' and 'Share Post' functionality by having a 'Dropdown List' in place of the current 'Copy Link' Button.

This has numerous benefits, some of them are mentioned👇🏻

Saves time as finding posts is a hassle for the average user. (Bookmarks)
Easy cross-platform sharing is free marketing and enhances user experience. (Share Post)
It'll be fun to peek into Balaji's Bookmarks :P (Project Idea?)

<img width="625" alt="Screenshot 2021-07-11 at 10 12 01 PM" src="https://user-images.githubusercontent.com/87209627/125204274-0f08ad80-e29a-11eb-9471-e8e82f97437f.png">

**Comment by lazynina** [comment]
Are you proposing the `bookmark` functionality as a consensus mechanism? If so, what advantages do you see to this? I think that would be pretty inefficient and isn't necessary to store on-chain. Perhaps this would be fine as a node-specific list of bookmarks.

A share post button makes sense, but we need a list of sites to which sharing is supported.  Additionally, mock ups would be appreciated to get a better sense of your idea.

**Reply by brootle** [reply]
> Are you proposing the `bookmark` functionality as a consensus mechanism? If so, what advantages do you see to this? I think that would be pretty inefficient and isn't necessary to store on-chain. Perhaps this would be fine as a node-specific list of bookmarks.
> 
> A share post button makes sense, but we need a list of sites to which sharing is supported. Additionally, mock ups would be appreciated to get a better sense of your idea.

@bluepartyhat please add "saved" to post field, after that I will add "Saved" tab and render posts there, just check what pots have "saved" as "true" for the logged user. Also add "save post" into three dots drop down menu, thanks!

@maebeam I think this can be done fairly easy, just need some copy/past/edit on backend side, but I have zero knowledge in Go lang
![image](https://user-images.githubusercontent.com/20090030/125755269-5ec85984-1027-47b3-9b56-b3c3067fea88.png)
![image](https://user-images.githubusercontent.com/20090030/125756141-8cc657cb-175e-401a-bf64-0859e0c6635d.png)

**Reply by cnzpll** [reply]
I vote for @brootle proposal

**Comment by mvanhalen** [comment]
Hi, for sharing posts I would suggest using the share api outlined [here](https://www.w3.org/TR/web-share/). This works great on mobile and major browsers support it fine as well on PC/Mac. Related to that is SEO discovery for posts and share metatags. But I will create a separate discussing for that.

**Reply by tijno** [reply]
@mvanhalen note that meta tags has been discussed in various issues / cips with a solution available at devclout/embedclout thats as yet not been implement on bitclout.com but is operational on tijn.club

**Reply by mvanhalen** [reply]
Ok cool. So that would solve the posts seo discovery? I saw the bitclout.com normal node is also missing a sitemap and robots.txt. I would expect something listing all the posts and users. I do see user profiles getting in Google though. For the post share I added a [PR](https://github.com/bitclout/frontend/pull/388) to open default share options of chrome, edge and safari. Which normally have all your favourite apps and a bunch of more things. Like favorite contacts and so on.
